### PR TITLE
allow routing to specific remote kernels

### PR DESCRIPTION
**Pull request by brettfo**
This will shortly be re-written when the composite kernel knows more about proxy kernel subkernels and commands, but it at least adds a useful test.

### CODE REVIEW

**Review by colombod — APPROVED** [review]
I was there

File: src/Microsoft.DotNet.Interactive/Extensions/KernelUriExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+
+namespace Microsoft.DotNet.Interactive.Extensions
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/Extensions/KernelUriExtensions.cs:6]: This namespace is intended to be about kernel extensions, not extension methods.

File: src/Microsoft.DotNet.Interactive/Connection/ProxyKernel.cs [hunk 2]
```diff
@@ -74,8 +75,10 @@ internal override async Task HandleAsync(KernelCommand command, KernelInvocation
                     return;
             }
 
+            var kernelUri = KernelUri.Parse(command.TargetKernelName);
+            var remoteTargetKernelName = kernelUri.GetRemoteKernelName();
             var targetKernelName = command.TargetKernelName;
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/Connection/ProxyKernel.cs:80]: Renaming this to `localTargetKernelName` might make it a bit more meaningful in context given these changes.

File: src/Microsoft.DotNet.Interactive/Extensions/KernelUriExtensions.cs (hunk 1 above)
Comment (brettfo) [reply src/Microsoft.DotNet.Interactive/Extensions/KernelUriExtensions.cs]: Moved to `Microsoft.DotNet.Interactive.Utility` alongside other extension methods.

**Review by jonsequitur — APPROVED** [review]

**Pull request merged**